code/modules/martial_arts/combos/sleeping_carp/keelhaul.dm

```diff
@@ -0,0 +1,19 @@
+/datum/martial_combo/sleeping_carp/keelhaul
+	name = "Keelhaul"
+	steps = list(MARTIAL_COMBO_STEP_HARM, MARTIAL_COMBO_STEP_GRAB)
+	explaination_text = "Kicks an opponent to the floor, knocking them down and dealing stamina damage to them!"
+
+/datum/martial_combo/sleeping_carp/keelhaul/perform_combo(mob/living/carbon/human/user, mob/living/target, datum/martial_art/MA)
+	user.do_attack_animation(target, ATTACK_EFFECT_KICK)
+	playsound(get_turf(target), 'sound/effects/hit_kick.ogg', 50, TRUE, -1)
+	target.apply_damage(10, BRUTE, BODY_ZONE_HEAD)
+	target.apply_damage(40, STAMINA)
+	if(!target.IsWeakened() && !target.resting && !target.stat)
+		target.visible_message("<span class='warning'>[user] kicks [target] in the head, sending them face first into the floor!</span>",
+						"<span class='userdanger'>You are kicked in the head by [user], sending you crashing to the floor!</span>")
+	else
+		target.visible_message("<span class='warning'>[user] kicks [target] in the head, leaving them reeling in pain!</span>",
+							"<span class='userdanger'>You are kicked in the head by [user], and you reel in pain!</span>")
+	target.Weaken(2)
+	add_attack_logs(user, target, "Melee attacked with martial-art [MA] : Keelhaul", ATKLOG_ALL)
+	return MARTIAL_COMBO_DONE
```

Comment:
Way want to re-look at this one, for consistency with TG---or look at what they had, pre-combat change, for ideas, as well.

Reply:
Right now, on TG. this knocks down for 4 seconds and deals 40 stamina damage if they are stood up. If they aren't stood up it disarms them and applies 40 stamina damage.  gitblame lists these changes as coming from a year ago. A short weaken and stamina damage seems to be the best I can do to replicate this? Maybe applying a slowdown and stamina damage, and then weakening if they are already slowed? I am not sure.